recipes/cmake/3.x.x/conanfile.py

```diff
@@ -37,8 +37,8 @@ def config_options(self):
             self.options.with_openssl = False
 
     def requirements(self):
-        if self.options.with_openssl:
-            self.requires("openssl/1.1.1t")
+        if self.options.get_safe("with_openssl", default=False):
+            self.requires("openssl/[>=1.1 <4]")
```

Comment:
I have tested that the oldest version in the recipe `3.19.8` is compatible with the most recent OpenSSL 3.x